src/Microsoft.ML.Core/Data/RowId.cs

```diff
@@ -9,109 +9,119 @@
 namespace Microsoft.ML.Runtime.Data
 {
     /// <summary>
-    /// A sixteen-byte unsigned integer.
+    /// A structure serving as a sixteen-byte unsigned integer. It is used as the row id of <see cref="IDataView"/>.
+    /// For datasets with millions of records, those IDs need to be unique, therefore the need for such a large structure to hold the values.
+    /// Those Ids are derived from other Ids of the previous components of the pipelines, and dividing the structure in two: high order and low order of bits,
+    /// and reduces the changes of those collisions even further.
     /// </summary>
-    public readonly struct UInt128 : IComparable<UInt128>, IEquatable<UInt128>
+    /// <seealso cref="Row.GetIdGetter"/>
+    public readonly struct RowId : IComparable<RowId>, IEquatable<RowId>
     {
-        // The low order bits. Corresponds to H1 in the Murmur algorithms.
-        public readonly ulong Lo;
-        // The high order bits. Corresponds to H2 in the Murmur algorithms.
-        public readonly ulong Hi;
+        ///<summary>The low order bits. Corresponds to H1 in the Murmur algorithms.</summary>
+        public readonly ulong LowerWord;
```

Comment:
I should have caught this in Ivan's suggestion, but the word `Word` shouldn't be used here. There is a certain connotation that "WORD" means 2 bytes - especially for Windows C++ developers.

My preference for these names is simply `Low` and `High`.